lumigator/python/mzai/jobs/inference/inference_config.py

```diff
@@ -30,10 +120,55 @@ class SamplingParameters(BaseModel):
     model_config = ConfigDict(extra="forbid")
 
 
+class PipelineConfig(BaseModel, arbitrary_types_allowed=True):
```

Comment:
Why do we allow arbitrary types?

Reply:
For the custom types that we have defined (i.e., `AssetPath`, `SupportedTask`, and `SerializableTorchDtype`). If I remember correctly it throws an error if I don't have this. I can test it again.

Reply:
Yes, this is what I'm getting if I remove it:

```
Job entrypoint command failed with exit code 1, last available logs (truncated to 20,000 chars):
  File "/tmp/ray/session_2024-12-10_02-57-03_358398_1/runtime_resources/pip/617697551215d8488f42ccab087d2364573ba842/virtualenv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 997, in match_type
    return self._unknown_type_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ray/session_2024-12-10_02-57-03_358398_1/runtime_resources/pip/617697551215d8488f42ccab087d2364573ba842/virtualenv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 515, in _unknown_type_schema
    raise PydanticSchemaGenerationError(
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class 'torch.dtype'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.10/u/schema-for-unknown-type
```

Reply:
Thanks for checking out! Yes, if I remember well we had the same issue in evaluator. What I appreciate here is that arbitrary types are allowed only within this config, instead of the basic class as in evaluator, which makes this version a bit more tight.